src/Data/RouteData.ts

```diff
@@ -1,112 +1,124 @@
-                        },
-                        {
-                            description: 'Six dots in a two-by-three grid',
-                            name: 'Braille',
-                            url: '/braille',
-                        },
-                        {
-                            description: 'Dots and dashes with separations',
-                            name: 'Morse Code',
-                            url: '/morse',
-                        },
-                        {
-                            description: 'Parts of a tic-tac-toe grid or X, some with dot',
-                            name: 'Pigpen',
-                            url: '/pigpen',
-                        },
-                        {
-                            description: ' Positions of two different flags',
-                            name: 'Semaphore',
-                            url: '/semaphore',
-                        },
-                    ],
-                    description: 'Different ways to represent letters',
-                    name: 'Encodings',
-                    rootUrl: '/encoding',
-                },
-                {
-                    children: [
-                        {
-                            description: 'Static display of number to letter mapping',
-                            name: 'Character Encodings',
-                            url: '/characterencodings',
-                        },
-                        {
-                            description: 'Words used to clearly identify a letter',
-                            name: 'NATO Alphabet',
-                            url: '/nato',
-                        },
-                        {
-                            description: 'Different simple flags which convey a letter',
-                            name: 'Naval Flags',
-                            url: '/navalflags',
-                        },
-                        {
-                            description: 'Find a resistor value using different colored bands',
-                            name: 'Resistors',
-                            url: '/resistors',
-                        }
-                    ],
-                    description: 'Miscellaneous information',
-                    name: 'Reference',
-                    rootUrl: '/reference',
-                }
-            ],
-            name: 'Puzztool',
-        };
-    }
-}
+export interface RouteChild {
+    description: string;
+    name: string;
+    url: string;
+}
+
+export interface RouteCategory {
+    children: RouteChild[];
+    description: string;
+    name: string;
+    rootUrl: string;
+}
+
+export interface RoutesInfo {
+    categories: RouteCategory[];
+    name: string;
+}
+
+export class RouteData {
+    public static getCategories() {
+        return RouteData.getRoutes().categories;
+    }
+
+    private static getRoutes(): RoutesInfo {
+        return {
+            categories: [
+                {

```

Comment:
I hate line endings.  This is the only diff section, confirmed via copy, paste, and Beyond Compare.

Reply:
Yeah.  Nothing here changed from above, just rebasing on top of master